src/MySqlConnector/MySqlConnection.cs

```diff
@@ -1062,22 +1064,10 @@ private async ValueTask<ServerSession> CreateSessionAsync(ConnectionPool? pool,
 				// only "fail over" and "random" load balancers supported without connection pooling
 				var loadBalancer = connectionSettings.LoadBalance == MySqlLoadBalance.Random && connectionSettings.HostNames!.Count > 1 ?
 					RandomLoadBalancer.Instance : FailOverLoadBalancer.Instance;
-
-				var session = new ServerSession(m_logger)
-				{
-					OwningConnection = new WeakReference<MySqlConnection>(this),
-				};
+				var session = await ServerSession.ConnectAndRedirectAsync(() => new ServerSession(m_logger), m_logger, null, connectionSettings, loadBalancer, this, null, startingTimestamp, null, actualIOBehavior, cancellationToken).ConfigureAwait(false);
```

Comment:
```suggestion
				var session = await ServerSession.ConnectAndRedirectAsync(() => new ServerSession(m_logger), m_logger, null, connectionSettings, loadBalancer, this, null, startingTimestamp, null, actualIOBehavior, connectToken).ConfigureAwait(false);
```

See line 1073 below.